src/encoded/static/components/item-pages/ExperimentSetView.js

```diff
@@ -550,87 +550,204 @@ export class OtherProcessedFilesStackedTableSectionPart extends React.PureCompon
+    };
+
+    constructor(props){
+        super(props);
+        this.toggleOpen = this.toggleOpen.bind(this);
+
+        this.state = {
+            'open' : props.defaultOpen
+        };
+    }
+
+    toggleOpen(e){
+        this.setState(function(oldState){
+            return { 'open' : !oldState.open };
+        });
+    }
+
+    render(){
+        const { files, width, href } = this.props;
+        const filesLen = files.length;
+        const passProps = _.extend({ width, href, files }, SelectedFilesController.pick(this.props));
+        return (
+            <div data-open={open} className="reference-files-section supplementary-files-section-part">
+                <h4 className="mb-15">
+                    <span className="inline-block clickable" onClick={this.toggleOpen}>
+                        <i className={"text-normal icon icon-fw icon-" + (open ? 'minus' : 'plus')} />
+                        { filesLen + " Reference File" + (filesLen > 1 ? "s" : "") }
+                    </span>
+                </h4>
+                <Collapse in={open} mountOnEnter>
+                    <div className="table-for-collection">
+                        <ProcessedFilesStackedTable {...passProps} files={files} />
+                    </div>
+                </Collapse>
+            </div>
+        );
+    }
+}
+
+
 /** This object renders the "Supplementary Files" section. */
-export class OtherProcessedFilesStackedTableSection extends React.PureComponent {
+export class SupplementaryFilesTabView extends React.PureComponent {
 
-    static checkOPFCollectionPermission(opfCollection){
-        return _.any(opfCollection.files || [], function(opf){
-            return opf && object.itemUtil.atId(opf) && !opf.error;
+    /** @returns {boolean} true if at least one file inside the collection is viewable */
+    static checkOPFCollectionPermission({ files }){
+        return _.any(files || [], function(file){
+            return file && object.itemUtil.atId(file) && !file.error;
         });
     }
 
+    /** @returns {boolean} true if at least one collection has at least one file inside the collection that is viewable */
     static checkOPFCollectionsPermissions(opfCollections){
-        return _.any(opfCollections || [], OtherProcessedFilesStackedTableSection.checkOPFCollectionPermission);
+        return _.any(opfCollections || [], SupplementaryFilesTabView.checkOPFCollectionPermission);
     }
 
-    static extendCollectionsWithExperimentFiles = memoize(function(context){
+    /**
+     * Combines files from ExperimentSet.other_processed_files and ExperimentSet.experiments_in_set.other_processed_files.
+     *
+     * @param {ExperimentSet} context - JSON response from server for this endpoint/page depicting an ExperimentSet.
+     * @returns {{ files: File[], title: string, type: string, description: string }[]} List of uniqued-by-title viewable collections.
+     */
+    static combinedOtherProcessedFiles = memoize(function(context){
```

Comment:
Logic of this function remains the same however was slightly refactored in attempt to improve readability